cmd/argocd/commands/app.go

```diff
@@ -1156,7 +1163,27 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 			argoSettings, err := settingsIf.Get(ctx, &settings.SettingsQuery{})
 			errors.CheckError(err)
 			diffOption := &DifferenceOption{}
-			if revision != "" {
+			if app.Spec.HasMultipleSources() && len(revisions) > 0 && len(sourceIndexes) > 0 {
+
+				revisionSourceMappings := make(map[int64]string, 0)
+				for i, index := range sourceIndexes {
+					if index < 0 {
+						errors.CheckError(fmt.Errorf("source-index cannot be less than or equal to 0. Index starts at 1."))
```

Comment:
People using ArgoCD are developers and are used to index starting at 0 IMO.

Reply:
I had initially started with 0 as the index, but we updated it to 1 as the cli users all might not be developers only.

Reply:
I too had a similar view. Something named Index, users may assume that it starts from 0. If we name the argument differntly like ordinal or position, may be we can emphasise that it starts from 1.

Reply:
Yeah I think consistency is more important for this PR, but I think this should be quickly reassessed before this it is used and released.